src/core/algorithms/fd/fd_algorithm.cpp

```diff
@@ -1,21 +1,22 @@
 #include "fd_algorithm.h"
 
-#include <map>
 #include <thread>
-#include <vector>
 
 #include "config/equal_nulls/option.h"
+#include "config/max_lhs/option.h"
 #include "config/tabular_data/input_table/option.h"
 
 namespace algos {
 
 FDAlgorithm::FDAlgorithm(std::vector<std::string_view> phase_names)
     : Algorithm(std::move(phase_names)) {
     RegisterOptions();
-    MakeOptionsAvailable({config::TableOpt.GetName(), config::EqualNullsOpt.GetName()});
+    MakeOptionsAvailable({config::MaxLhsOpt.GetName(), config::TableOpt.GetName(),
```

Comment:
`max_lhs` should be made available in the execute stage of the algorithm, not in the data load stage (in the current code it is made available in the `MakeExecuteOptsAvailable`, not when the algorithm object is constructed). Not sure if it's a desirable behaviour overall, but it is how it's implemented right now, so for this PR let's leave `max_lhs` as execute stage option